tests/Requests/RequestTest.php

```diff
@@ -5,220 +5,115 @@
-        $this->assertSame('DE', $productRequests[1]->getLanguage()->getCode());
-        $this->assertSame('2019-11-15', $productRequests[1]->getRequestedDeadline()->format('Y-m-j'));
-
-        $this->assertNull($productRequests[2]->getAcceptedDeadline());
-        $this->assertSame('NA', $productRequests[2]->getFormatCode());
-        $this->assertNull($productRequests[2]->getInternalProductReference());
-        $this->assertSame('ENV', $productRequests[2]->getStatusCode());
-        $this->assertFalse($productRequests[2]->isTrackChanges());
-        $this->assertSame('CS', $productRequests[2]->getLanguage()->getCode());
-        $this->assertSame('2019-04-15', $productRequests[2]->getRequestedDeadline()->format('Y-m-j'));
-
-        $this->assertNull($productRequests[3]->getAcceptedDeadline());
-        $this->assertSame('NA', $productRequests[3]->getFormatCode());
-        $this->assertNull($productRequests[3]->getInternalProductReference());
-        $this->assertSame('ENV', $productRequests[3]->getStatusCode());
-        $this->assertFalse($productRequests[3]->isTrackChanges());
-        $this->assertSame('FR', $productRequests[3]->getLanguage()->getCode());
-        $this->assertSame('2019-11-15', $productRequests[3]->getRequestedDeadline()->format('Y-m-j'));
-
-        $this->assertNull($productRequests[4]->getAcceptedDeadline());
-        $this->assertSame('NA', $productRequests[4]->getFormatCode());
-        $this->assertNull($productRequests[4]->getInternalProductReference());
-        $this->assertSame('ENV', $productRequests[4]->getStatusCode());
-        $this->assertFalse($productRequests[4]->isTrackChanges());
-        $this->assertSame('BG', $productRequests[4]->getLanguage()->getCode());
-        $this->assertNull($productRequests[4]->getRequestedDeadline());
-
-        // Assert reference.
-        $reference = $return->getReference();
-        $this->assertSame(1738, $reference->getId());
-        $this->assertSame('MYREF-DEF-123456', $reference->getInternalReference());
-        $this->assertNull($reference->getInternalTechnicalId());
-        $this->assertSame(2249, $reference->getNumber());
-        $this->assertSame(0, $reference->getPart());
-        $this->assertSame('TRA', $reference->getProductType());
-        $this->assertSame('DGT', $reference->getRequesterCode());
-        $this->assertSame(0, $reference->getVersion());
-        $this->assertSame(2018, $reference->getYear());
 
-        // Assert status code.
-        $this->assertSame('ENV', $return->getStatusCode());
+        $values = [
+            'response' => $response,
+        ];
+
+        foreach ($expectations['assertions'] as $type => $expectationsTypes) {
+            foreach ($expectationsTypes as $expression => $expectation) {
+                $this->{$type}(
+                    $expectation,
+                    $this->expressionLanguage->evaluate(
+                        $expression,
+                        $values
```

Comment:
We can actually evaluate expressions without the need of PHPUnit methods.

Reply:
Fixed.